jbpm-case-mgmt/jbpm-case-mgmt-impl/src/main/java/org/jbpm/casemgmt/impl/model/instance/CaseFileInstanceImpl.java

```diff
@@ -152,8 +152,12 @@ public void removeAll() {
     }
 
     @Override
-    public Object getData(String name) {
-        return this.data.get(name);
+    public Object getData(String name) {        
+        Object result = this.data.get(name);
+        if (result == null && "caseid".equalsIgnoreCase(name)) {
+            result = getCaseId();
```

Comment:
What happens if I define my custom caseid as a case file item? Would I break it? And can't we call CaseFileInstance#getCaseId() method for that?

Reply:
it won't break, this is just a shortcut mainly for CMMN that do not define "local" variables and only case file items. The idea here is to allow to use caseId as expressions #{caseId} for tasks names, descriptions, data inputs etc.

Reply:
OK, thanks.